zig-dxread/src/main.zig

```diff
@@ -0,0 +1,72 @@
+const std = @import("std");
+const fs = std.fs;
+const gpa = std.heap.GeneralPurposeAllocator(.{}){};
+const allocator = gpa.allocator;
+const ArrayList = std.ArrayList;
+const erorset_1 = error{OutOfMemory};
+
+const RECORD_SIZE = 80 * 384;
+
+const DX_header = struct {
+    year: u32,
+    month: u32,
+    day: u32,
+    utc: u32,
+    satid: u32,
+    sattyp: u32,
+    nchans: u32,
+    nghtimg: u32,
+};
+
+const DX_data = struct {
+    header: DX_header,
+};
+
+
+pub fn readHeader(buffer: [RECORD_SIZE]u8) !DX_header {
```

Comment:
These functions can be moved to their struct namespaces (as you can see in my example refactor file)

Reply:
ah yes definitely. neat thanks!